### PR TITLE
fixing error of not handled enum AUTOPLAY in switch

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Fixes https://github.com/brave/brave-browser/issues/5750

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/5998) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [X] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
After https://github.com/brave/brave-browser/pull/5995 is merged or cherry-picked:
run npm run init -- --target_os=android --target_arch=x64
run npm run build -- --target_os=android --target_arch=x64

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/android_webview-browser-aw_permission_manager.cc.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+diff --git a/android_webview/browser/aw_permission_manager.cc b/android_webview/browser/aw_permission_manager.cc
+index f1979fdcc078f31d8caad5da6cd68906b138f263..cc68cf801b05615590f187123bb9aafd30521518 100644
+--- a/android_webview/browser/aw_permission_manager.cc
++++ b/android_webview/browser/aw_permission_manager.cc
+@@ -335,6 +335,7 @@ int AwPermissionManager::RequestPermissions(
+       case PermissionType::BACKGROUND_FETCH:
+       case PermissionType::IDLE_DETECTION:
+       case PermissionType::PERIODIC_BACKGROUND_SYNC:
++      case PermissionType::AUTOPLAY:
```
Comment (bridiver) [inline patches/android_webview-browser-aw_permission_manager.cc.patch]: patches should almost always use `define` to make them extensible. See `permission_controller_impl.cc`, but is this the right place to put it? We do have permission requests for autoplay afaik cc @darkdh

Comment (darkdh) [reply patches/android_webview-browser-aw_permission_manager.cc.patch]: On desktop, we will convert `PermissionType::AUTOPLAY` to `CONTENT_SETTINGS_TYPE_AUTOPLAY` (`PermissionTypeToContentSetting`) and then have a general way to handle it through content setting. On Android, it seems like we have to specifically define and implement virtual function in `AwBrowserPermissionRequestDelegate` for each support type.

Comment (AlexeyBarabash) [reply patches/android_webview-browser-aw_permission_manager.cc.patch]: @bridiver @darkdh 
Updated the patch to use defines.

As for special processing `PermissionType::AUTOPLAY`, this `AwPermissionManager::RequestPermissions` eventually goes to WebViewContentsClientAdapter.onPermissionRequest (https://cs.chromium.org/chromium/src/android_webview/glue/java/src/com/android/webview/chromium/WebViewContentsClientAdapter.java?l=562) , and then if WebView consumer had set up the `WebChromeClient`,  ```mWebChromeClient.onPermissionRequest(adapter)``` will be invoked. Owner of `WebChromeClient` can then call `permissionRequest.grant` or `permissionRequest.deny`.

Currently Chromium WebView files aw_*.* are included into the build, but it is not available to set as a system webview, also it is hard to believe, that android app developers who use WebView would expect the Brave WebView  would be a system WebView and apps have some special handling for `AUTOPLAY` permission.

So `AUTOPLAY` is permissions in this patch remains denied.

**Comment by AlexeyBarabash** [comment]
seeing CI  
```
Version mismatch between brave-browser (android_build_autoplay_enum/0.72.69) and 
brave-core (android_build_autoplay_enum/0.72.60) in package.json
```
done rebase

**Comment by AlexeyBarabash** [comment]
@bridiver could you please re-approve , as I had to rebase to make CI succeed.

**Pull request merged**